pandas-stubs/_typing.pyi

```diff
@@ -53,36 +53,126 @@ from pandas.core.dtypes.dtypes import (
 
 from pandas.io.formats.format import EngFormatter
 
-# `Incomplete` is equivalent to `Any`. Use it to annotate symbols that you don't
-# know the type of yet and that should be changed in the future. Use `Any` only
-# where it is the only acceptable type.
-Incomplete: TypeAlias = Any
+P = ParamSpec("P")
+
+HashableT = TypeVar("HashableT", bound=Hashable)
+HashableT1 = TypeVar("HashableT1", bound=Hashable)
+HashableT2 = TypeVar("HashableT2", bound=Hashable)
+HashableT3 = TypeVar("HashableT3", bound=Hashable)
+HashableT4 = TypeVar("HashableT4", bound=Hashable)
+HashableT5 = TypeVar("HashableT5", bound=Hashable)
+
+# array-like
 
 ArrayLike: TypeAlias = ExtensionArray | np.ndarray
 AnyArrayLike: TypeAlias = ArrayLike | Index[Any] | Series[Any]
+
+# list-like
+
+_T_co = TypeVar("_T_co", covariant=True)
+
+class SequenceNotStr(Protocol[_T_co]):
+    @overload
+    def __getitem__(self, index: SupportsIndex, /) -> _T_co: ...
+    @overload
+    def __getitem__(self, index: slice, /) -> Sequence[_T_co]: ...
+    def __contains__(self, value: object, /) -> bool: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[_T_co]: ...
+    def index(self, value: Any, start: int = ..., stop: int = ..., /) -> int: ...
+    def count(self, value: Any, /) -> int: ...
+    def __reversed__(self) -> Iterator[_T_co]: ...
+
+ListLike: TypeAlias = AnyArrayLike | SequenceNotStr | range
```

Comment:
Not sure, does SequenceNotStr need a paramter here? Something like `ListLike: TypeAlias = AnyArrayLike | SequenceNotStr[Any] | range`

Reply:
I think you're right.  Will fix that.